packages/mobile/src/verify/VerificationEducationScreen.tsx

```diff
@@ -1,205 +1,156 @@
-import colors from '@celo/react-components/styles/colors'
-import fontStyles from '@celo/react-components/styles/fonts'
-import { componentStyles } from '@celo/react-components/styles/styles'
-import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
+import TextButton from '@celo/react-components/components/TextButton.v2'
+import colors from '@celo/react-components/styles/colors.v2'
+import fontStyles from '@celo/react-components/styles/fonts.v2'
+import { Spacing } from '@celo/react-components/styles/styles.v2'
+import { StackScreenProps, useHeaderHeight } from '@react-navigation/stack'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
 import { ScrollView, StyleSheet, Text, View } from 'react-native'
-import Modal from 'react-native-modal'
-import SafeAreaView from 'react-native-safe-area-view'
-import { connect } from 'react-redux'
+import { useSafeArea } from 'react-native-safe-area-view'
+import { useDispatch } from 'react-redux'
 import { ErrorMessages } from 'src/app/ErrorMessages'
 import ErrorMessageInline from 'src/components/ErrorMessageInline'
-import { Namespaces, withTranslation } from 'src/i18n'
+import i18n, { Namespaces } from 'src/i18n'
 import { setHasSeenVerificationNux } from 'src/identity/actions'
 import { isUserBalanceSufficient } from 'src/identity/utils'
 import { INVITE_FEE } from 'src/invite/saga'
-import { nuxNavigationOptions } from 'src/navigator/Headers'
-import { navigate, navigateHome } from 'src/navigator/NavigationService'
+import { HeaderTitleWithSubtitle, nuxNavigationOptions } from 'src/navigator/Headers.v2'
+import { navigateHome } from 'src/navigator/NavigationService'
 import { Screens } from 'src/navigator/Screens'
-import { RootState } from 'src/redux/reducers'
+import { TopBarTextButton } from 'src/navigator/TopBarButton.v2'
+import { StackParamList } from 'src/navigator/types'
+import useSelector from 'src/redux/useSelector'
+import VerificationLearnMoreDialog from 'src/verify/VerificationLearnMoreDialog'
+import VerificationSkipDialog from 'src/verify/VerificationSkipDialog'
 
 const VERIFICATION_FEE_ESTIMATE = Number(INVITE_FEE) * 0.9
 
-interface DispatchProps {
-  setHasSeenVerificationNux: typeof setHasSeenVerificationNux
-}
-
-interface StateProps {
-  userBalance: string | null
-}
+type ScreenProps = StackScreenProps<StackParamList, Screens.VerificationEducationScreen>
 
-type Props = WithTranslation & DispatchProps & StateProps
+type Props = ScreenProps
 
-interface State {
-  isModalVisible: boolean
-}
+function VerificationEducationScreen({ route, navigation }: Props) {
+  const showSkipDialog = route.params?.showSkipDialog || false
+  const [showLearnMoreDialog, setShowLearnMoreDialog] = useState(false)
+  const { t } = useTranslation(Namespaces.onboarding)
+  const userBalance = useSelector((state) => state.stableToken.balance)
+  const balanceIsSufficient = isUserBalanceSufficient(userBalance, VERIFICATION_FEE_ESTIMATE)
```

Comment:
i don't think this is quite expensive enough to be worth memoizing, just thought this was an interesting read (https://react-redux.js.org/next/api/hooks#using-memoizing-selectors) and we could reselect here if we wanted

Reply:
Yes thanks well aware of this 👍